src/openloco/windows/town_list.cpp

```diff
@@ -1,16 +1,1581 @@
+        {
+            return (x / 112) + (y / 112) * 5;
+        }
+
+        // 0x0049AEFD
+        static void on_scroll_mouse_down(ui::window* self, int16_t x, int16_t y, uint8_t scroll_index)
+        {
+            int16_t xPos = (x / 112);
+            int16_t yPos = (y / 112) * 5;
+            auto index = getRowIndex(x, y);
+
+            for (auto i = 0; i < self->var_83C; i++)
+            {
+                auto rowInfo = self->row_info[i];
+                index--;
+                if (index < 0)
+                {
+                    self->row_hover = rowInfo;
+
+                    if (self->current_tab == common::widx::tab_build_misc_buildings - common::widx::tab_town_list)
+                        _lastSelectedMiscBuilding = static_cast<uint8_t>(rowInfo);
+                    else
+                        _lastSelectedBuilding = static_cast<uint8_t>(rowInfo);
+
+                    updateBuildingColours(self);
+
+                    int32_t pan = (self->width >> 1) + self->x;
+                    loc16 loc = { xPos, yPos, static_cast<int16_t>(pan) };
+
+                    audio::play_sound(audio::sound_id::click_down, loc, pan);
+                    self->saved_view.mapX = -16;
+                    dword_1135C34 = 0x80000000;
+                    byte_1135C65 = 0;
+                    self->invalidate();
+                    break;
+                }
+            }
+        }
+
+        // 0x0049AEBA
+        static void on_scroll_mouse_over(ui::window* self, int16_t x, int16_t y, uint8_t scroll_index)
+        {
+            auto index = getRowIndex(x, y);
+            uint16_t rowInfo = y;
+            auto i = 0;
+            for (; i < self->var_83C; i++)
+            {
+                rowInfo = self->row_info[i];
+                index--;
+                if (index < 0)
+                {
+                    self->var_846 = rowInfo;
+                    self->invalidate();
+                    break;
+                }
+            }
+        }
+
+        // 0x0049ABC5
+        static void on_close(window* self)
+        {
+            if (input::is_tool_active(self->type, self->number))
+                input::cancel_tool();
+        }
+
+        // 0x0049AEA1
+        static void event_08(window* self)
+        {
+            if (self->var_846 != 0xFFFF)
+            {
+                self->var_846 = 0xFFFF;
+                self->invalidate();
+            }
+        }
+
+        // 0x0049B206
+        static void updateBuildingList(window* self)
+        {
+            auto buildingCount = 0;
+            for (auto i = 0; i < 128; i++)
+            {
+                auto buildingObj = objectmgr::get<building_object>(i);
+                if (buildingObj == nullptr)
+                    continue;
+                if (self->current_tab == common::widx::tab_build_misc_buildings - common::widx::tab_town_list)
+                {
+                    if ((buildingObj->flags & building_object_flags::misc_building) == 0)
```

Comment:
I much prefer the ! syntax over a compare with 0. I find it much harder to read but I'm not sure what the convention in the codebase tends to be.